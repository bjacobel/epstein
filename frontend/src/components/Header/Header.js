import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

import {
  title,
  header,
  hotlinks,
  hotlink,
  mobileBurger,
  burgerLines,
  active,
  mobileHotlinks,
} from './style.css';
import { blacklink } from '../../stylesheets/shared.css';

export const MenuButton = ({ toggleMenu, menuActive }) => {
  return (
    <button
      type="button"
      aria-label="expand mobile menu"
      className={mobileBurger}
      onClick={toggleMenu}
    >
      <span className={classNames(burgerLines, { [active]: menuActive })} />
    </button>
  );
};

export default () => {
  const [menuActive, toggleMenuActive] = useState(false);
  const menuCloseOnClick = () => {
    if (menuActive) toggleMenuActive(false);
  };
  return (
    <nav className={header}>
      <h1 className={title}>
        <Link className={blacklink} to="/">
          epstein.flights
        </Link>
      </h1>
      <MenuButton
        toggleMenu={() => toggleMenuActive(!menuActive)}
        aria-haspopup="true"
        aria-controls="menu-list"
        aria-expanded={menuActive} // @TODO: aria-expanded should be false when css viewport is big
        menuActive={menuActive}
      />
      <ul className={classNames(hotlinks, { [mobileHotlinks]: menuActive })}>
        <li className={hotlink} tabIndex="-1">
          <Link className={blacklink} to="/about" onClick={menuCloseOnClick}>
            About
          </Link>
        </li>
        <li className={hotlink} tabIndex="-1">
          <Link className={blacklink} to="/search" onClick={menuCloseOnClick}>
            Search
          </Link>
        </li>
      </ul>
    </nav>
  );
};
