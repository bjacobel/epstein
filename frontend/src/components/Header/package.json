{
  "main": "./Header.js",
  "private": true
}
