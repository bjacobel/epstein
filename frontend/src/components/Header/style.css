@value mobile, margin from stylesheets/measurements;
@value black from stylesheets/colors;

.header {
  display: flex;
  width: 100%;
  border-bottom: 1px solid black;
  align-items: center;
  height: 75px;
  margin-bottom: margin;
}

.title {
  composes: robotoMono from '../../stylesheets/shared';
  font-weight: 300;
  flex-grow: 1;
}

.hotlink {
  composes: roboto from '../../stylesheets/shared';
  padding: 0 margin;
}

.hotlinks {
  list-style-type: none;
  display: flex;
  align-items: center;

  @media (max-width: mobile) {
    display: none;
  }

  &.mobileHotlinks {
    border: 1px solid black;
    display: block;
    padding: margin;
    position: absolute;
    background-color: white;
    right: margin;
    margin: 0;
    top: 75px;

    .hotlink {
      padding: 0.5em;
    }
  }
}

.mobileBurger {
  border: none;
  box-shadow: none;
  outline: none;
  z-index: 2147483647;
  display: none;
  width: 45px;
  height: 45px;
  padding: 10px 5px;
  cursor: pointer;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);

  @media (max-width: mobile) {
    display: block;
  }

  .burgerLines {
    width: 35px;
    height: 3px;
    display: block;
    position: relative;
    background-color: black;
    top: 0;
    right: 0;
    opacity: 1;
    user-select: none;
    transition: all 0.25s ease;

    &::before {
      content: '\00A0';
      height: 3px;
      display: block;
      position: relative;
      background-color: black;
      user-select: none;
      transition: all 0.25s ease;
      top: -9px;
    }

    &::after {
      content: '\00A0';
      height: 3px;
      display: block;
      position: relative;
      background-color: black;
      user-select: none;
      transition: all 0.25s ease;
      top: 6px;
    }

    &.active {
      background: transparent;

      &::before {
        transform: rotate3d(0, 0, 1, -45deg);
        background-color: black;
        top: 3px;
      }

      &::after {
        transform: rotate3d(0, 0, 1, 45deg);
        background-color: black;
        top: 0;
      }
    }
  }
}
