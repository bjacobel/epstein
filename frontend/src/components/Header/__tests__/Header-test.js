import React from 'react';
import { mount } from 'enzyme';

import Header, { MenuButton } from '../Header';

describe('header component', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when tapped', () => {
    const wrapper = mount(<Header />);
    wrapper.find(MenuButton).simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
