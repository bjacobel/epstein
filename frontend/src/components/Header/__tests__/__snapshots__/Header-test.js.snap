// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component matches snapshot 1`] = `
<_default>
  <nav
    className="header"
  >
    <h1
      className="title"
    >
      <Link
        className="blacklink"
        to="/"
      >
        <a
          href="/"
        >
          epstein.flights
        </a>
      </Link>
    </h1>
    <MenuButton
      aria-controls="menu-list"
      aria-expanded={false}
      aria-haspopup="true"
      menuActive={false}
      toggleMenu={[Function]}
    >
      <button
        aria-label="expand mobile menu"
        className="mobileBurger"
        onClick={[Function]}
        type="button"
      >
        <span
          className="burgerLines"
        />
      </button>
    </MenuButton>
    <ul
      className="hotlinks"
    >
      <li
        className="hotlink"
        tabIndex="-1"
      >
        <Link
          className="blacklink"
          onClick={[Function]}
          to="/about"
        >
          <a
            href="/about"
          >
            About
          </a>
        </Link>
      </li>
      <li
        className="hotlink"
        tabIndex="-1"
      >
        <Link
          className="blacklink"
          onClick={[Function]}
          to="/search"
        >
          <a
            href="/search"
          >
            Search
          </a>
        </Link>
      </li>
    </ul>
  </nav>
</_default>
`;

exports[`header component matches snapshot when tapped 1`] = `
<_default>
  <nav
    className="header"
  >
    <h1
      className="title"
    >
      <Link
        className="blacklink"
        to="/"
      >
        <a
          href="/"
        >
          epstein.flights
        </a>
      </Link>
    </h1>
    <MenuButton
      aria-controls="menu-list"
      aria-expanded={true}
      aria-haspopup="true"
      menuActive={true}
      toggleMenu={[Function]}
    >
      <button
        aria-label="expand mobile menu"
        className="mobileBurger"
        onClick={[Function]}
        type="button"
      >
        <span
          className="burgerLines active"
        />
      </button>
    </MenuButton>
    <ul
      className="hotlinks mobileHotlinks"
    >
      <li
        className="hotlink"
        tabIndex="-1"
      >
        <Link
          className="blacklink"
          onClick={[Function]}
          to="/about"
        >
          <a
            href="/about"
          >
            About
          </a>
        </Link>
      </li>
      <li
        className="hotlink"
        tabIndex="-1"
      >
        <Link
          className="blacklink"
          onClick={[Function]}
          to="/search"
        >
          <a
            href="/search"
          >
            Search
          </a>
        </Link>
      </li>
    </ul>
  </nav>
</_default>
`;
